src/cluster.ts

```diff
@@ -399,34 +401,36 @@ Please use the format 'my-cluster' or '${instance.name}/clusters/my-cluster'.`);
    * region_tag:bigtable_cluster_set_meta
    */
   setMetadata(
-    metadata: CreateClusterOptions,
+    metadata: SetClusterMetadataOptions,
     gaxOptionsOrCallback?: CallOptions | SetClusterMetadataCallback,
     cb?: SetClusterMetadataCallback
   ): void | Promise<SetClusterMetadataResponse> {
     const callback =
       typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
     const gaxOptions =
-      typeof gaxOptionsOrCallback === 'object' && gaxOptionsOrCallback
+      typeof gaxOptionsOrCallback === 'object'
         ? gaxOptionsOrCallback
         : ({} as CallOptions);
 
     const reqOpts: ICluster = {
       name: this.name,
+      serveNodes: metadata.nodes,
```

Comment:
This is where I think we should pass all user-provided input straight through, as opposed to the whitelist approach where we only recognize certain parameters.

Reply:
(mapping `metadata.nodes` to `serveNodes` is still good)

Reply:
If applicable, user will need to pass the full
`location` as full path in the form of
`projects/<project>/locations/<zone_id>`

and a `defaultStorageType` (not `storage`) in upper case characters.

Done